File: misc/aooswd/ex/bool_ex_interpreter/cpp/symbol_table.cpp
#include "symbol_table.h"

void SymbolTable::setValue (const string& name, Value* v)
{
  int index = getIndexOf (name);
  table[index].setValue(v);
}

Value* SymbolTable::getValue (const string& name)
{
  int index = getIndexOf(name);
  return table[index].getValue();
}

int SymbolTable::getIndexOf (const string& name)
{
  int index = 0;
  while (index < table.size() && table[index].getName() != name)
    index++;
  return index;
}

File: misc/aooswd/ex/bool_ex_interpreter/java/SymbolTable.java
import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;
import java.util.Vector;

/**
 * @author dgayler
 *
 */
public class SymbolTable
{

	private Vector table;

	public SymbolTable()
	{
		table = new Vector();
	}

	public void insert(SymbolData data)
	{
		table.addElement((data));
	}

	public void setValue(String name, Object value)
	{
		int index = getIndexOf(name);
		((SymbolData) table.elementAt(index)).setValue(value);
	}

	public Object getValue(String name)
	{
			int index = getIndexOf(name);
			SymbolData data = (SymbolData) table.elementAt(index);
			return data.getValue();
	}

	private int getIndexOf(String name)
	{
		int index = 0;
		while (index < table.size()
			&& !name.equals(((SymbolData) table.elementAt(index)).getName()))
			index++;
		return index;
	}

}
